axelor-supplychain/src/main/java/com/axelor/apps/supplychain/db/repo/SaleOrderSupplychainRepository.java

```diff
@@ -40,6 +44,14 @@ public SaleOrder copy(SaleOrder entity, boolean deep) {
     copy.setDeliveryState(DELIVERY_STATE_NOT_DELIVERED);
     copy.setAmountInvoiced(null);
     copy.setStockMoveList(null);
+
+    for (SaleOrderLine saleOrderLine : saleOrderLines) {
+      saleOrderLine.setDeliveryState(null);
+      saleOrderLine.setDeliveredQty(null);
```

Comment:
I think some another fields should be also reseted like invoice amount information.